### ISSUE TITLE
Compatibility with ROOT 6.24.00 in M1 Silicon MacBook

### ISSUE THREAD

**Issue by kratsg** [histfitter/histfitter#21]
In GitLab by @eballabe on May 3, 2021, 11:09

Dear HistFitter experts,

ROOT v6.24.00 says to come with up to 4x speed improvements and additional robustness so I thought to give it a try. 
I've tried to install HF locally on my laptop which is based on arm64, but I get some compilation errors (not sure if related to ROOT or libSusyFitter.so). I've attached the compilation log.

Cheers,
Eric

[compilation_log.out](/uploads/1f7b7655f6c2277cd0861fc3da8b76f6/compilation_log.out)

**Comment by kratsg** [comment]
In GitLab by @longjon on May 18, 2021, 06:09

Hi @eballabe 

I'm not very familiar with compiling on different architectures or ROOT on macs, but based on your log file:

`Undefined symbols for architecture arm64:
  "ROOT::Math::normal_cdf(double, double, double)", referenced from:
      RooStats::get_Pvalue(RooStats::HypoTestInverterResult const*, bool) in StatTools.o`

it sounds like your ROOT installation is missing some libraries.  Did you build ROOT yourself?  I'd guess either the libraries not available for arm, or perhaps you missed enabling some options when compiling ROOT (if you did it yourself), e.g. minuit2.

Can you use any of the ROOT functions/classes that the linker is complaining about with your ROOT install?

Cheers,
Jonathan

**Comment by kratsg** [comment]
In GitLab by @eballabe on May 31, 2021, 09:53

Hi @longjon 

(sorry for the late reply)

My HF compilation test was made in an arm lapton with macOS BigSur Version 11.3.
I had a conda environment in the terminal, and ROOT v6.24.00 was installed with the command:
> conda install -c conda-forge root

I also checked that this was happening in a terminal without the conda environment, but with ROOT v6.24.00 pre-compiled binary distribution, setup this way:
> source /Applications/root_v6.24.00/bin/thisroot.sh 

However, looking here (https://root.cern/releases/release-62400/), it seems that the latest ROOT version is compatible with macOS 11.2, but presumably not with macOS 11.3, which is the OS I'm currently using. 

Given this mismatch between the ROOT version available and my OS version, I'm closing this issue for the moment, with the plan of repeating the test when a new ROOT version will be available.

I tested HF compilation in a x86 laptop with macOS BigSur Version 11.2.3 and with the latest compatible ROOT v6.24.00, and it compiled fine.

Cheers,
Eric
